browser/ui/color/brave_color_mixer.cc

```diff
@@ -688,15 +688,21 @@ void AddBraveOmniboxLightThemeColorMixer(ui::ColorProvider* provider,
                                          const ui::ColorProviderKey& key) {
   ui::ColorMixer& mixer = provider->AddMixer();
 
+  auto pick_color = [&](leo::Color color) {
+    if (!key.custom_theme) {
+      return leo::GetColor(color, leo::Theme::kLight);
+    }
+
+    return PickColorContrastingToOmniboxResultsBackground(
+        key, mixer, leo::GetColor(color, leo::Theme::kLight),
+        leo::GetColor(color, leo::Theme::kDark));
+  };
   mixer[kColorBraveOmniboxResultViewSeparator] = {
-      leo::GetColor(leo::Color::kColorDividerSubtle, leo::Theme::kLight)};
+      pick_color(leo::Color::kColorDividerSubtle)};
+  mixer[kColorBravePlayerActionViewBorder] = {
+      pick_color(leo::Color::kColorDividerSubtle)};
```

Comment:
@simonhong , I refactored this part a little bit

Reply:
Looks good 👍🏼